tests/Tests/Properties/Substrings.hs

```diff
@@ -231,6 +233,47 @@ tl_isSuffixOf s   = L.isSuffixOf s`eqP` TL.isSuffixOf (packS s)
 t_isInfixOf s     = L.isInfixOf s `eqP` T.isInfixOf (packS s)
 tl_isInfixOf s    = L.isInfixOf s `eqP` TL.isInfixOf (packS s)
 
+-- | Generator for substrings that keeps the element order.
+-- Aka: "1234567890" -> "245680"
+genOrdSubseq :: T.Text -> Gen T.Text
+genOrdSubseq txt =
+  T.pack . transform <$> genTransformMap
```

Comment:
I can't understand what this implementation is actually doing, but is this not all we need? 

```suggestion
genOrdSubseq txt = 
  T.pack . map fst . filter snd . zip (T.unpack txt) <$> vectorOf (T.length txt) arbitrary :: Gen [Bool]
```
We're just trying to select a random sample of elements from the given string in order right? Are there any other properties we need from this? Trying to ensure that the result is non-empty isn't too hard, you just generate two lists of Bools:
```haskell
```suggestion
genOrdSubseq txt = do
    firstLen <- choose (1,T.length txt - 2)
    prefix <-  vectorOf firstLen arbitrary
    suffix <- vectorOf (T.length txt - firstLen - 1) arbitrary
    pure $ T.pack . map fst . filter snd . zip (T.unpack txt) $ prefix ++ True : suffix
```
(not sure if I have the maths right, but I'm trying to make a list of `length txt` which has a `True` somewhere in the middle)